From 15bc66f90d5c510c8f85e6da77078670b11ab8d0 Mon Sep 17 00:00:00 2001
From: Liu Hao <lh_mouse@126.com>
Date: Mon, 29 Jan 2018 18:24:58 +0800
Subject: [PATCH 1/2] include/usbioctl.h: Add required headers for
 `__C89_NAMELESS` and Windows data types.

This caused trouble when <usbioctl.h> was included after <windows.h>.

Reference: https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/content/usbioctl/ne-usbioctl-_usb_connection_status
Signed-off-by: Liu Hao <lh_mouse@126.com>
---
 mingw-w64-headers/include/usbioctl.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mingw-w64-headers/include/usbioctl.h b/mingw-w64-headers/include/usbioctl.h
index 57498b50..47f510ed 100644
--- a/mingw-w64-headers/include/usbioctl.h
+++ b/mingw-w64-headers/include/usbioctl.h
@@ -15,6 +15,8 @@
 #ifndef __USBIOCTL_H__
 #define __USBIOCTL_H__
 
+#include <_mingw_mac.h>
+#include <minwindef.h>
 #include <winapifamily.h>
 
 #if WINAPI_FAMILY_PARTITION (WINAPI_PARTITION_DESKTOP)
@@ -251,7 +253,7 @@ typedef struct _USB_NODE_CONNECTION_INFORMATION_EX {
   USB_CONNECTION_STATUS ConnectionStatus;
   USB_PIPE_INFO PipeList[0];
 } USB_NODE_CONNECTION_INFORMATION_EX,*PUSB_NODE_CONNECTION_INFORMATION_EX;
-;
+
 #if _WIN32_WINNT >= 0x0600
 typedef union _USB_HUB_CAP_FLAGS {
   ULONG ul;
-- 
2.16.1

